File: Path Planning/Pathfinder/Scenario.cs
using System.Drawing;

namespace Pathfinder
{
    /// <summary>
    /// Holds a specific graph, with start and destination, and with a scenario name.
    /// </summary>
    public class Scenario
    {
        public string Name { get; private set; }
        public GridGraph Graph { get; private set; }
        public Point Start { get; private set; }
        public Point Destination { get; set; }
        public Scenario(string name, GridGraph g, Point start, Point destination)
        {
            Name = name;
            Graph = g;
            Start = start;
            Destination = destination;
        }
        public override string ToString()
        {
            return Name;
        }
    }
}

File: Path Planning/PathfinderVB/Scenario.vb
Imports System.Drawing

Namespace Pathfinder
    Public Class Scenario
        Public Property Name As String
        Public Property Graph As GridGraph
        Public Property Start As Point
        Public Property Destination As Point

        Public Sub New(ByVal n As String, ByVal g As GridGraph, ByVal strt As Point, ByVal dest As Point)
            Name = n
            Graph = g
            Start = strt
            Destination = dest
        End Sub

        Public Overrides Function ToString() As String
            Return Name
        End Function
    End Class
End Namespace

